### PR TITLE
adding more instructions for enabling dicom cast

### PR DESCRIPTION
**Pull request by moiradillon12**
## Description
Adding example end to end flow for how to enable auth for dicom cast with managed identity

## Related issues
Addresses [AB#76021](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/76021)
[AB#75894](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/75894)

### CODE REVIEW

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 1]
```diff
@@ -1,12 +1,90 @@
...
+
+1. [Deploy a Medical Imaging Server for DICOM](../quickstarts/deploy-via-azure.md)
+1. [Deploy a FHIR Server](https://github.com/microsoft/fhir-server)
+1. [Deploy DICOM Cast](../quickstarts/deploy-dicom-cast.md)
+
+## Configure Authentication using Managed Identity
+
+Currently there are three types of authentication supported for both Azure API for FHIR and Medical Imaging Server for DICOM: Managed Identity, OAuth2 Client Credential and OAuth2 User Password. The authentication can be configured via the application settings by the appropriate values in the `Authentication` property of the given server. For details on the three types, see [DICOM Cast authentication](/converter/dicom-cast/docs/authentication.md).
+
+This section will provide an end to end guide for configuring authentication with Managed Identity.
```
Comment (smithago) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:23]: Maybe I am not understanding this. I am assuming this documentation is to support MI for dicom-cast. The ACI already has the system assigned MI. That should be given access to Fhir and Dicom to use MI, correct?

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 2]
```diff
@@ -1,12 +1,90 @@
...
+1. Select **Secrets** in the menu bar and click **Generate/Import**. Use the tables below to add secrets for your DICOM and FHIR servers. For each secret, use the **Manual Upload option** and click **Create**:
+
+#### Medical Imaging Server for DICOM Secrets
+
+| Name | Value |
+| :------- | :----- |
+| DicomWeb--Endpoint | ```<dicom-server-url>``` |
+| DicomWeb--Authentication--Enabled | "true" |
+| DicomWeb--Authentication--AuthenticationType | "ManagedIdentity" |
+| DicomWeb--Authentication--ManagedIdentityCredential--Resource | ```<dicom-server-url>``` |
```
Comment (smithago) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:64]: Is this tested? The code is also looking for tenantId.

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md (hunk 1 above)
Comment (moiradillon12) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:23]: Thanks @smithago I didn't realize ACI already has MI assigned, I thought this was required to enable MI. @jackliums could you please take a look at this? Thanks!

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md (hunk 2 above)
Comment (moiradillon12) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:64]: I got this configuration from here: https://github.com/microsoft/dicom-server/blob/master/converter/dicom-cast/src/Microsoft.Health.DicomCast.Hosting/appsettings.json

@brandonpollett  do we need tenantID added here too

Comment (brandonpollett) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:64]: The code passes through the tenantId if it's set on the configuration, but it isn't required. The method it calls `GetAccessTokenAsync` takes the parameters as `string tenantId = null,`

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md (hunk 1 above)
Comment (brandonpollett) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:23]: So the general flow would be 

1. Deploy the DICOM Server and FHIR Server
1. Deploy DICOM Cast (the template defaults to having managed identity enabled on the ACI)
1. Update your auth in the DICOM Server or FHIR server to allow access to the MI
1. Update your DICOM Cast configuration to use the managed identity

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 3]
```diff
@@ -1,12 +1,90 @@
...
+    1. Under **Configure from template**, select **Secret Management**.
+    1. Under **Select principal**, click **None selected**. Search for your Service Principle, click **Select** and then **Add**. 
+    1. Select **Save**.
+1. Select **Secrets** in the menu bar and click **Generate/Import**. Use the tables below to add secrets for your DICOM and FHIR servers. For each secret, use the **Manual Upload option** and click **Create**:
+
+#### Medical Imaging Server for DICOM Secrets
+
+| Name | Value |
+| :------- | :----- |
+| DicomWeb--Endpoint | ```<dicom-server-url>``` |
```
Comment (brandonpollett) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:61]: If you're using the template, this value and the `Fhir--Endpoint" are already set.

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 4]
```diff
@@ -1,12 +1,90 @@
...
+| DicomWeb--Authentication--Enabled | "true" |
+| DicomWeb--Authentication--AuthenticationType | "ManagedIdentity" |
+| DicomWeb--Authentication--ManagedIdentityCredential--Resource | ```<dicom-server-url>``` |
+
+#### FHIR Server Secrets
+
+| Name | Value |
+| :------- | :----- |
+| fhir--Endpoint | ```<fhir-server-url>``` |
+| fhir--Authentication--Enabled | "true" |
```
Comment (brandonpollett) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:71]: I don't know if these are case sensitive, but we should go ahead and update the `fhir` -> `Fhir`

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 5]
```diff
@@ -1,12 +1,90 @@
...
+1. [Deploy a FHIR Server](https://github.com/microsoft/fhir-server)
+1. [Deploy DICOM Cast](../quickstarts/deploy-dicom-cast.md)
+
+## Configure Authentication using Managed Identity
+
+Currently there are three types of authentication supported for both Azure API for FHIR and Medical Imaging Server for DICOM: Managed Identity, OAuth2 Client Credential and OAuth2 User Password. The authentication can be configured via the application settings by the appropriate values in the `Authentication` property of the given server. For details on the three types, see [DICOM Cast authentication](/converter/dicom-cast/docs/authentication.md).
+
+This section will provide an end to end guide for configuring authentication with Managed Identity.
+
+### Create a resource application for FHIR and DICOM servers
```
Comment (brandonpollett) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:25]: We can probably just link to setting up auth in each of the repos?

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 6]
```diff
@@ -1,12 +1,90 @@
...
+1. Select **Azure Active Directory** > **App Registrations** > **New registration**:
+    1. Enter a **Name** for your app registration.
+    2. In **Redirect URI**, select **Web** and enter the **URL** of your App Service.
+    3. Select **Register**.
+1. Select **Expose an API** > **Set**. You can specify a URI as the **URL** of your app service or use the generated App ID URI. Select **Save**.
+1. Select **Add a Scope**:
+    1. In **Scope name**, enter *user.impersonation*.
+    1. In the text boxes, add an admin consent display name and admin consent description you want users to see on the consent page. For example, *access my app*.
+
+### Set the Authentication for your FHIR & DICOM App Services
```
Comment (brandonpollett) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:39]: Same for this? Seems like the instructions already exist elsewhere?

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md [hunk 7]
```diff
@@ -1,12 +1,90 @@
...
+1. Select **Configuration** to update the **Audience** and **Authority**:
+    1. Set the **Application ID URI** from the App Service as the **Audience**.
+    1. **Authority** is whichever tenant your application exists in, for example: ```https://login.microsoftonline.com/<tenant-name>.onmicrosoft.com```.
+
+### Update Key Vault for DICOM Cast
+
+1. Navigate to the DICOM Cast Key Vault that was created when you deployed DICOM Cast.
+1. Select **Access Policies** in the menu bar and click **Add Access Policy**.
+    1. Under **Configure from template**, select **Secret Management**.
+    1. Under **Select principal**, click **None selected**. Search for your Service Principle, click **Select** and then **Add**.
```
Comment (brandonpollett) [inline docs/how-to-guides/sync-dicom-metadata-to-fhir.md:53]: Nit: you're not looking for a "Service Principle" here. You're updating keyvault to allow yourself to manage the secrets. We could potentially just link to: https://docs.microsoft.com/en-us/azure/key-vault/general/assign-access-policy-portal

File: docs/how-to-guides/sync-dicom-metadata-to-fhir.md (hunk 3 above)
Comment (moiradillon12) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:61]: They don't show up automatically in Secrets, is that ok?

Comment (brandonpollett) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:61]: Yea, they're set as environment variables on the ACI instance.

Comment (moiradillon12) [reply docs/how-to-guides/sync-dicom-metadata-to-fhir.md:61]: Great thanks!

**Review by StevenBorg — APPROVED** [review]
Walked through end-to-end to confirm it works. Painful, but it works.

**Pull request merged**